src/controllers/scenario/index.ts

```diff
@@ -3,37 +3,46 @@ import {
   Get,
   Path,
   Route,
-} from "tsoa";
+} from 'tsoa'
 import config from 'config'
 import { exec } from 'child_process'
-import { ExamplesResult } from '../../../types'
+import { ExamplesResult, IScenario, HostResponse, Executables } from '../../../types'
 import Logger from '../../utils/Logger'
 
-let log = Logger.child({ controller: '/scenario', ...config.get('morello') });
-const { address, username } = config.get('morello')
-
 @Route('scenario')
-export class scenario extends Controller {
-  @Get('{id}')
-  public async get(@Path() id: string): Promise<ExamplesResult> {
-    const cheriBin = `cat ./binaries/${id}-cheri | ssh ${username}@${address} "cat >${id}; chmod +x ${id}; ${id} with args; rm ${id}`
-    return ({
-      self: await new Promise((resolve) => {
-        log.info(`executing ${id} example on self`);
+export class scenario extends Controller implements IScenario {
+  address: string
+  port: number
+  log: typeof Logger
+  
+  constructor() {
+    super()
+    this.address = `${config.get('morello.username')}@${config.get('morello.address')}`
+    this.port = config.get('morello.port')
+    this.log = Logger.child({ controller: '/scenario', ...config.get('morello') })
+  }
 
-        exec(`./binaries/${id}-aarch64`, (err, stdout, stderr) => resolve({
-          status: (err || stderr) ? 'error': 'success',
-          output: err ? err : stderr ? stderr : stdout,
-        }))
-      }),
-      morello: await new Promise((resolve) => {
-        log.info(`executing ${id} exampole on the morello host`);
+  execute(bin: string): Promise<HostResponse> {
+    const scp = `scp -P ${this.port} bin/${bin} ${this.address}:/tmp`
```

Comment:
we should probably have some validation to make sure that the file exists. For now this si fine

Reply:
Going to create another story to add a binary file validatioon after we have done publish/package/build oir whatever we will decide, if you do not mind?